csrc/python_frontend/fusion_definition.cpp

```diff
@@ -673,4 +674,19 @@ std::vector<std::pair<double, double>> FusionDefinition::getValTolerances(
   return get_val_constants(preschedFusion(), inputs);
 }
 
+int64_t FusionDefinition::setupSegmentation(
+    const at::ArrayRef<c10::IValue>& inputs) {
+  NVF_CHECK(id().has_value(), "FusionDefinition definition does not exist!");
+  NVF_ERROR(
+      segmentation_state_ == nullptr, "SegmentationState already exists!");
+  segmentation_state_ = std::make_unique<SegmentationState>();
+  return segmentation_state_->setupSegmentation(
+      preschedFusion(), map_value_to_fid_, inputs);
+}
+
+void FusionDefinition::finalizeSegmentation() {
+  // Destroy SegmentedState
+  segmentation_state_.reset();
+}
```

Comment:
a design question. What's the point of separating this into two steps?

Doesn't look like anything else is happening between `setupSegmentation()` to `finalizeSegmentation()`.

Reply:
https://github.com/NVIDIA/Fuser/pull/3335 adds the `buildSegmentation()` to translate the CPP segments to their corresponding python definition.

We still need to destroy the `segmentation_state_` in this PR.